geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala

```diff
@@ -239,4 +242,42 @@ object TileLayer {
       case NOCOMPRESSION => NoCompression
       case DEFLATECOMPRESSION => DeflateCompression
     }
+
+  def combineBands[K: ClassTag, L <: TileLayer[K]: ClassTag](
+    sc: SparkContext,
+    layers: ArrayList[L]
+  ): RDD[(K, MultibandTile)] = {
+    val scalaLayers = layers.asScala.toArray
+
+    val rdds: Array[RDD[(K, MultibandTile)]] =
+      scalaLayers.map { case (v: L) => v.rdd }
+
+    val arr = Array.ofDim[RDD[(K, (Int, MultibandTile))]](rdds.size)
+
+    for ((layer, index) <- rdds.zipWithIndex) {
+      arr(index) = layer.mapValues { (index, _) }
+    }
+
+    val unioned = sc.union(arr.toSeq)
+
+    val bands: RDD[(K, Map[Int, Vector[Tile]])] =
+      unioned.combineByKey(
```

Comment:
There should be some kind of check for error or warning if the layouts of the different layers do not match. Thats a kind of error thats easy to do with a surprising result.

Reply:
I had the layouts checked in python: https://github.com/jbouffard/geopyspark/blob/acb4163cbbcfdd8ad6dc3adceef5a9c6bf96effc/geopyspark/geotrellis/__init__.py#L66

Reply:
Gotcha